plugins/newrelic/src/api/index.test.ts

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NewRelicClient } from '.';
+import { FetchApi } from '@backstage/core-plugin-api';
+import { DiscoveryApi } from '@backstage/core-plugin-api';
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('NewRelicClient', () => {
+  test.each([
+    ['https://test.test/BASEPATH/apm/api/applications.json', '/BASEPATH'],
+    ['https://test.test/BASEPATH2/apm/api/applications.json', '/BASEPATH2'],
+    ['https://test.testBASEPATH3/apm/api/applications.json', 'BASEPATH3'],
+    ['https://test.test/newrelic/apm/api/applications.json', undefined],
+  ])(
+    'It correctly forms the request url (%p) when proxyPathBase is %p',
+    async (expectedUrl, basePathOverride) => {
+      const mockedDiscoveryApi: DiscoveryApi = {
+        getBaseUrl: jest.fn().mockResolvedValueOnce('https://test.test'),
+      };
+
+      const mockedFetchApi: FetchApi = {
```

Comment:
Wow, thanks for the nice test! Love the effort.

One possible improvement if you feel up for it - I'm almost afraid to suggest it - we've centered a lot on `msw` instead of mocking when it comes to network API testing, [see for example here](https://github.com/backstage/backstage/blob/cc14e1ba0ec638dbc9c07636e8db89e3cfd2e5cc/packages/backend-common/src/reading/AzureUrlReader.test.ts#L172). Does that sound compelling or would you prefer to merge as-is?

To be clear, you can still keep the injection of `fetchApiRef`, that's still very good practice either way!

Reply:
I don't believe that I have used ```msw``` before, but I would like to give it a try to learn a new way of doing this. Thanks for the example! 👍